core/src/main/java/io/confluent/rest/RestConfig.java

```diff
@@ -199,8 +199,8 @@
       "ssl.endpoint.identification.algorithm";
   protected static final String SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_DOC =
       "The endpoint identification algorithm to validate the server hostname using the "
-      + "server certificate. Leave blank to use Jetty's default.";
-  protected static final String SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_DEFAULT = "";
+      + "server certificate. Default algorithm is https.";
+  protected static final String SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_DEFAULT = "https";
```

Comment:
Another option is to support previous behavior i.e disable the host verification and allow users to override.

Reply:
I think it is better to enable hostname verification as in the current PR since we already did this for Kafka clients and disabling it is insecure. We need to make sure we include this change in the upgrade notes though.